dnf5/commands/system-upgrade/system-upgrade.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "system-upgrade.hpp"
+
+#include "../offline/offline.hpp"
+
+#include <libdnf5-cli/utils/userconfirm.hpp>
+#include <libdnf5/base/goal.hpp>
+#include <libdnf5/conf/const.hpp>
+#include <libdnf5/conf/option_path.hpp>
+#include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
+#include <libdnf5/utils/fs/file.hpp>
+#include <sys/wait.h>
+#include <systemd/sd-journal.h>
+
+#include <iostream>
+#include <string>
+
+using namespace libdnf5::cli;
+
+namespace dnf5 {
+
+void SystemUpgradeCommand::pre_configure() {
+    throw_missing_command();
+}
+
+void SystemUpgradeCommand::set_parent_command() {
+    auto * arg_parser_parent_cmd = get_session().get_argument_parser().get_root_command();
+    auto * arg_parser_this_cmd = get_argument_parser_command();
+    arg_parser_parent_cmd->register_command(arg_parser_this_cmd);
+    arg_parser_parent_cmd->get_group("subcommands").register_argument(arg_parser_this_cmd);
+}
+
+void SystemUpgradeCommand::set_argument_parser() {
+    get_argument_parser_command()->set_description("Prepare system for upgrade to a new release");
+}
+
+void SystemUpgradeCommand::register_subcommands() {
+    register_subcommand(std::make_unique<SystemUpgradeDownloadCommand>(get_context()));
+    register_subcommand(std::make_unique<OfflineCleanCommand>(get_context()));
+    register_subcommand(std::make_unique<OfflineRebootCommand>(get_context()));
+    register_subcommand(std::make_unique<OfflineLogCommand>(get_context()));
+}
+
+SystemUpgradeSubcommand::SystemUpgradeSubcommand(Context & context, const std::string & name)
+    : Command(context, name),
+      datadir(std::filesystem::path{libdnf5::SYSTEM_STATE_DIR} / "system-upgrade"),
+      state(datadir / "state.toml") {}
```

Comment:
Is this different from the filename defined in the `libdnf5::offline` on purpose?

Reply:
Ah no, this was some old code, thanks.